server/src/main/java/org/elasticsearch/action/admin/cluster/bootstrap/TransportGetDiscoveredNodesAction.java

```diff
@@ -124,4 +125,16 @@ public String toString() {
             });
         }
     }
+
+    private static boolean checkWaitRequirements(GetDiscoveredNodesRequest request, Set<DiscoveryNode> nodes) {
```

Comment:
The extra logic here needs corresponding tests in `TransportGetDiscoveredNodesActionTests`.